core/services/bulletprooftxmanager/errors.go

```diff
@@ -75,6 +76,20 @@ func (s *sendError) IsTemporarilyUnderpriced() bool {
 	return s != nil && s.err != nil && s.Error() == parLimitReached
 }
 
+func (s *sendError) IsInsufficientEth() bool {
+	if s == nil || s.err == nil {
+		return false
+	}
+	switch s.Error() {
+	// Geth
+	case "insufficient funds for transfer", "insufficient funds for gas * price + value", "insufficient balance for transfer":
+		return true
+	default:
+		// Parity
+		return parInsufficientEth.MatchString(s.Error())
+	}
```

Comment:
There is a method to the madness.

Geth uses hardcoded strings for its error messages. Regex is not required because we can match the string exactly and I would rather avoid unnecessary regexes.

Parity requires regexes because it adds additional numbers and interpolated values into its error messages.

I copied the same pattern in this function as you can see in `isFatalSendError` at the bottom of this file.

I'd be reluctant to make it inconsistent since errors are already so messy to handle anyway.

Reply:
makes sense :+1: